src/main/scala/sangria/relay/Connection.scala

```diff
@@ -170,30 +158,59 @@ object Edge {
 
 case class DefaultEdge[T](node: T, cursor: String) extends Edge[T]
 
-case class PageInfo(
+trait PageInfo {
+  def hasNextPage: Boolean
+  def hasPreviousPage: Boolean
+  def startCursor: Option[String]
+  def endCursor: Option[String]
+}
+
+case class DefaultPageInfo(
   hasNextPage: Boolean = false,
   hasPreviousPage: Boolean = false,
   startCursor: Option[String] = None,
-  endCursor: Option[String] = None)
+  endCursor: Option[String] = None) extends PageInfo
 
 object PageInfo {
-  def empty = PageInfo()
+  def empty: DefaultPageInfo = DefaultPageInfo()
+
+  def apply(
+    hasNextPage: Boolean = false,
+    hasPreviousPage: Boolean = false,
+    startCursor: Option[String] = None,
+    endCursor: Option[String] = None
+  ): PageInfo = DefaultPageInfo(hasNextPage, hasPreviousPage, startCursor, endCursor)
+
+  def defaultPageInfoType[Ctx, P <: PageInfo : ClassTag]: ObjectType[Ctx, P] =
```

Comment:
Maybe we can move it in the companion object of `DefaultPageInfo`, WDYT?

Reply:
Also, do you need `P` type parameter here? Wouldn't be enough to define it as `ObjectType[Ctx, PageInfo]`?

Reply:
I think so, and I moved it in the companion object of `DefaultPageInfo`.  
`P` is not necessary for `DefaultPageInfo` with default `PageInfo`, but I need it for `DefaultPageInfo` with `P`.